lib/rubygems/uninstaller.rb

```diff
@@ -376,4 +378,32 @@ def safe_delete(&block)
 
     raise e
   end
+
+  private
+
+  def announce_deletion_of(spec)
+    name = spec.full_name
+    say "Successfully uninstalled #{name}"
+    if default_spec_matches?(spec)
+      say(
+        "There was both a regular copy and a default copy of #{name}. The " \
+          "regular copy was successfully uninstalled, but the default copy " \
+          "was left around because default gems can't be removed."
+      )
+    end
+  end
+
+  # @return true if the specs of any default gems are `==` to the given `spec`.
+  def default_spec_matches?(spec)
+    !default_specs_that_match(spec).empty?
+  end
+
+  # @return [Array] specs of default gems that are `==` to the given `spec`.
+  def default_specs_that_match(spec)
+    default_specs = []
+    Gem::Specification.each_spec([Gem.default_specifications_dir]) do |s|
+      default_specs << s
+    end
+    default_specs.select {|default_spec| spec == default_spec }
```

Comment:
Maybe we could save default specs to an instance variable when they are first listed at the beginning, and use it here so that we don't need to look for them every time?